### DISCUSSION TITLE
Authorization in type field takes effect after the query

### DISCUSSION THREAD

**Discussion by Fainaru** [MichalLytek/type-graphql#1587] [Q&A]
Hello, I would like to know how I can make the @Authorized in a type field act before the query, since for example I have some things that I want the users to see and others not, and that it works for each item in the array and column of this same one is very heavy since if I have 20 columns and 1000 rows I would use the Authorized function 20000 times, and if it were before I would only do it once

**Comment by MichalLytek** [comment]
We use on production the GraphQL Yoga solution:
https://the-guild.dev/graphql/envelop/plugins/use-generic-auth

AFAIR it's calculated from the query tree and evaluated before execution.

**Reply by Fainaru** [reply]
Thank you very much for answering and I did not understand well, are you telling me then that instead of using Authorized(), I should use use-generic-auth instead?

**Reply by Fainaru** [reply]
because with the authorized this is what happens to me
<img width="287" alt="Captura de pantalla 2023-12-22 a la(s) 18 38 12" src="https://github.com/MichalLytek/type-graphql/assets/72484366/da6f8928-7b2d-41d5-ac58-98ae5ea3a17e">
<img width="433" alt="Captura de pantalla 2023-12-22 a la(s) 18 42 29" src="https://github.com/MichalLytek/type-graphql/assets/72484366/c080ef16-a329-4410-adfa-a41062477d46">
<img width="716" alt="Captura de pantalla 2023-12-22 a la(s) 18 46 06" src="https://github.com/MichalLytek/type-graphql/assets/72484366/76c7a349-4a36-4c10-9d91-4b9b5e909656">

It gives me all that for the array and I only have it in the id, if I had it in all of them there would be more, that's why I'm talking about validating it before and not after, I don't know if it can be done or if I'm explaining myself correctly

**Reply by MichalLytek** [reply]
Yes, you should use other solution. The one built-in in TypeGraphQL is simple and basic, does not cover all needs.

**Reply by Fainaru** [reply]
and in the case of InputTypes, since authorization and using generic authentication doesn't work, could you use a custom class validator? For example, I have a "where", I want some to be able to access some parameters of the input and others not.
